android/java/org/chromium/chrome/browser/crypto_wallet/fragments/MarketFragment.java

```diff
@@ -0,0 +1,185 @@
+import java.util.List;
+
+public class MarketFragment extends Fragment implements CoinMarketsCallback {
+    private static final String TAG = "MarketFragment";
+
+    private LockableNestedScrollView mRootView;
+    private RecyclerView mCoinMarkets;
+    private ShimmerFrameLayout mShimmerViewContainer;
+    private LinearLayout mSingleItemShimmer;
+    private CardView mShimmerCardView;
+    private LinearLayout mShimmerListContainer;
+    private CardView mMarketListCardView;
+
+    private MarketCoinAdapter mMarketCoinAdapter;
+
+    private WalletModel mWalletModel;
+    private MarketModel mMarketModel;
+
+    private boolean mShimmerActive = true;
+
+    public static MarketFragment newInstance() {
+        return new MarketFragment();
+    }
+
+    @Override
+    public void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setHasOptionsMenu(true);
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            Api33AndPlusBackPressHelper.create(
+                    this, (FragmentActivity) requireActivity(), () -> requireActivity().finish());
+        }
+        try {
+            BraveActivity activity = BraveActivity.getBraveActivity();
+            mWalletModel = activity.getWalletModel();
+            mMarketModel = mWalletModel.getMarketModel();
+        } catch (BraveActivity.BraveActivityNotFoundException exception) {
+            Log.e(TAG, "onCreate", exception);
+        }
+        mMarketCoinAdapter = new MarketCoinAdapter(getContext(), mMarketModel);
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
+            @Nullable Bundle savedInstanceState) {
+        mRootView = (LockableNestedScrollView) inflater.inflate(
+                R.layout.fragment_market, container, false);
+        mRootView.setScrollingEnabled(false);
+        mShimmerViewContainer = mRootView.findViewById(R.id.shimmer_view_container);
+        mSingleItemShimmer = mRootView.findViewById(R.id.market_coin_item);
+        mShimmerCardView = mRootView.findViewById(R.id.card_view_market_placeholder);
+        mShimmerListContainer = mRootView.findViewById(R.id.shimmer_list);
+        mMarketListCardView = mRootView.findViewById(R.id.market_list_card_view_view_container);
+
+        mCoinMarkets = mRootView.findViewById(R.id.coin_markets_recycler_view);
+        mCoinMarkets.addItemDecoration(
+                new DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL));
+        mCoinMarkets.setAdapter(mMarketCoinAdapter);
+        mCoinMarkets.setLayoutManager(new LinearLayoutManager(getActivity()));
+        mMarketModel.getCoinMarkets(this);
+
+        final int visibleShimmerItems =
+                getVisibleShimmerItems(mSingleItemShimmer, mShimmerCardView);
+        for (int i = 1; i < visibleShimmerItems; i++) {
+            LayoutInflater.from(requireContext())
```

Comment:
nit: `LayoutInflater.from(requireContext())` can be moved outside for and reused in loop here and below.

Reply:
Yes, nice catch! I can reuse the `inflater` method parameter.